test/e2e/kubectl.go

```diff
@@ -188,6 +195,114 @@ var _ = Describe("Kubectl client", func() {
 			if e, a := "hi", execOutput; e != a {
 				Failf("Unexpected kubectl exec output. Wanted %q, got %q", e, a)
 			}
+
+			By("executing a command in the container through an HTTP proxy")
+			// Start by checking for the environment variable before anything else
+			kubectlBinPath := os.Getenv("KUBECTLBIN")
+			// Fail if the variable isn't set
+			if kubectlBinPath == "" {
+				Failf("KUBECTLBIN environment variable must be set to the full path of the kubectl binary")
+			}
+
+			apiserverAddr := os.Getenv("APISERVERADDR")
+			// Fail if the variable isn't set
+			if apiserverAddr == "" {
+				Failf("APISERVERADDR environment variable must be set to the full address to the api server")
+			}
+
+			// start the proxy container
+			goproxyPodPath := filepath.Join(testContext.RepoRoot, "test/images/goproxy/pod.yaml")
+			runKubectl("create", "-f", goproxyPodPath, fmt.Sprintf("--namespace=%v", ns))
+
+			// start exec-proxy-tester container
+			netexecPodPath := filepath.Join(testContext.RepoRoot, "contrib/for-tests/netexec/pod.yaml")
+			runKubectl("create", "-f", netexecPodPath, fmt.Sprintf("--namespace=%v", ns))
+			checkPodsRunningReady(c, ns, []string{goproxyContainer, netexecContainer}, podStartTimeout)
+
+			// get the proxy address
+			goproxyPod, err := c.Pods(ns).Get(goproxyContainer)
+			if err != nil {
+				Failf("Unable to get the goproxy pod. Error: %s", err)
+			}
+			proxyAddr := fmt.Sprintf("http://%s:8080", goproxyPod.Status.PodIP)
+			// get the exec address
+			netexecPod, err := c.Pods(ns).Get(netexecContainer)
+			netexecAddr := fmt.Sprintf("http://%s:8080", netexecPod.Status.PodIP)
+
+			// upload kubectl to remote exec server
+			postBodyBuffer := &bytes.Buffer{}
+			postBodyWriter := multipart.NewWriter(postBodyBuffer)
+
+			// Set up the form file
+			fileWriter, err := postBodyWriter.CreateFormFile("file", "kubectl")
+			if err != nil {
+				Failf("Unable to to write kubectl at %s to buffer. Error: %s", kubectlBinPath, err)
+			}
+
+			// Open up the kubectl binary for reading
+			fh, err := os.Open(kubectlBinPath)
+			if err != nil {
+				Failf("Unable to to access kubectl at %s. Error: %s", kubectlBinPath, err)
+			}
+
+			// Copy kubectl binary into the file writer
+			_, err = io.Copy(fileWriter, fh)
+			if err != nil {
+				Failf("Unable to to copy kubectl at %s into the file writer. Error: %s", kubectlBinPath, err)
+			}
+			// Nothing more should be written to this instance of the postBodyWriter
+			postBodyWriter.Close()
+
+			resp, err := http.Post(fmt.Sprintf("%s/upload", netexecAddr), postBodyWriter.FormDataContentType(), postBodyBuffer)
+
+			uploadBinaryName, err := ioutil.ReadAll(resp.Body)
```

Comment:
Check `err` first. If it's non-nil, `Failf`. Next check `resp.StatusCode`. Finally, read the body.